### ISSUE TITLE
Running Gradle with --no-daemon still forks a daemon when org.gradle.jvmargs are specified

### ISSUE THREAD

**Issue by esycat** [gradle/gradle#1434]
To avoid OOM exceptions we specify the following in our gradle.properties when running the daemon: `org.gradle.jvmargs=-XX:MaxPermSize=256m -Xms256m -Xmx512m`

However, when we don't want to run the daemon (f.e. because there's a plugin doing `System.console` stuff which doesn't work in daemon-style) we should specify `--no-daemon` or put `org.gradle.daemon=false` in our `gradle.properties`.

However when doing either of the above settings a daemon is still forked:
```
Note: in order to honour the org.gradle.jvmargs and/or org.gradle.java.home values specified for this build, it is necessary to fork a new JVM.
This forked JVM is effectively a single-use daemon process. In order to avoid the slowdown associated with this extra process, you might want to consider running Gradle with --daemon.
```

Basically, there's no way to not run Gradle as a daemon when there's `org.gradle.jvmargs` specified. This is quite annoying.

### Context
Migrated from [GRADLE-2357](https://issues.gradle.org/browse/GRADLE-2357) with 19 votes.

**Comment by oehme** [comment]
Gradle has to start a new process because you specified memory requirements that the client JVM did not satisfy. If you want Gradle to run the build inside the client VM, then that client VM needs to satisfy the memory requirements. You can ensure that by modifying your `GRADLE_OPTS`.
